File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def solution(n, computers):
    
    def dfs(i,computers,visited):
        visited[i] = 1
        
        for j in range(n):
            if not visited[j] and computers[i][j] :
                dfs(j,computers,visited)     
        return 
    
    answer = 0
    visited = [0]*n
    
    for i in range(n):
        if not visited[i] :
            dfs(i,computers,visited)
            answer += 1
            
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    public int[] visited;
    public int solution(int n, int[][] computers) {
        visited = new int[n];
        int answer=0;
        
        for(int i=0; i<n; i++){
            if(visited[i]==0){
                answer++;
                dfs(i,n, computers);   
            }
        }
        return answer;
    }
    
    public void dfs(int node, int n, int[][] computers){
        visited[node]=1;
                
        for(int j=0; j<n;j++){
            if(computers[node][j]==1 && visited[j]==0){
                dfs(j,n, computers);
            }
        }
    }
}